miden-lib/src/faucets/mod.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{assembler::assembler, auth::AuthScheme};
+use miden_objects::{
+    accounts::{Account, AccountCode, AccountId, AccountStorage, AccountType, AccountVault},
+    assembly::ModuleAst,
+    assets::TokenSymbol,
+    crypto::merkle::MerkleStore,
+    utils::{string::ToString, vec},
+    AccountError, Felt, StarkField, Word, ZERO,
+};
+
+const MAX_MAX_SUPPLY: u64 = 1 << 63;
+const MAX_DECIMALS: u8 = 18;
+
+/// Creates a new faucet account with basic faucet interface, specified authentication scheme,
+/// and provided meta data (token symbol, decimals, max supply).
+///
+/// The basic faucet interface exposes two procedures:
+/// - `distribute`, which mints an assets and create a note for the provided recipient.
+/// - `burn`, which burns the provided asset.
+///
+/// `distribute` requires authentication. The authentication procedure is defined by the specified
+/// authentication scheme. `burn` does not require authentication and can be called by anyone.
+///
+/// Public key information for the scheme is stored in the account storage at slot 0. The token
+/// metadata is stored in the account storage at slot 1.
+pub fn create_basic_faucet(
+    init_seed: [u8; 32],
+    symbol: TokenSymbol,
+    decimals: u8,
+    max_supply: Felt,
+    auth_scheme: AuthScheme,
+) -> Result<(Account, Word), AccountError> {
+    // Atm we onlt have RpoFalcon512 as authentication scheme and this is also the default in the
+    // faucet contract, so we can just use the public key as storage slot 0.
+    let auth_data: Word = match auth_scheme {
+        AuthScheme::RpoFalcon512 { pub_key } => pub_key.into(),
+    };
```

Comment:
when we add more auth methods, it will make sense to add a trait for this, instead of having a `AuthScheme`.

that way a user (e.g. a wallet), can implement a new scheme by implementing the trait. it also simplifies this code (by moving the handling of the data somewhere else).

with that said, I don't think we have to do it on this PR, just bringing it up for the future.